### ISSUE TITLE
Error when using default value in Redshift create table

### ISSUE THREAD

**Issue by Clepech** [sqlfluff/sqlfluff#3150]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Parsing error on create table statement containing column with default value

### Expected Behaviour

Linter should success

### Observed Behaviour

Linter fails:
```
= [foo.sql] FAIL                                                                                                                                                               
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE
                       | TABLE staging_layer.tmp_daily_ad_...'
```

### How to reproduce

Run lint command
```bash
sqlfluff lint --dialect=redshift foo.sql
```

Where `foo.sql` contains this query:

```sql
CREATE TABLE some_table
(
    some_column INTEGER NOT NULL DEFAULT 1
);

```

### Dialect

Redshift

### Version

0.12.0

### Configuration

[tool.sqlfluff.core]

encoding = "utf8"
rules = "L001,L002,L003"

[tool.sqlfluff.indentation]
indented_joins = false
template_blocks_indent = false

[tool.sqlfluff.rules]
max_line_length = 120
allow_scalar = false
comma_style = "trailing"

[tool.sqlfluff.rules.L010]  # Keywords
# L020 & L030 inherit from this rule
capitalisation_policy = "upper"

[tool.sqlfluff.templater.jinja]
library_path="mock_tempates"

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
Technically [by the docs](https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_TABLE_NEW.html), this should be written as:

```sql
CREATE TABLE some_table
(
    some_column INTEGER DEFAULT 1 NOT NULL
);
```

But I've seen a few discrepancies with the Redshift docs as to what they say they support and what they actually support.

Have loosened the ordering in #3151 to allow both.
